backend/stakepoold/server.go

```diff
@@ -381,45 +380,65 @@ func (ctx *appContext) loadData() {
 	defer ctx.Unlock()
 }
 
-func (ctx *appContext) processWinningTickets(wt WinningTicketsForBlock) {
-	blockHash := wt.blockHash
-	blockHeight := wt.blockHeight
+func (ctx *appContext) sendTickets(blockHash, ticket *chainhash.Hash, msa string, height int64) error {
+	sstx, err := walletCreateVote(ctx, blockHash, height, ticket, msa)
+	if err != nil {
+		return fmt.Errorf("failed to create vote: %v", err)
+	}
+
+	_, err = nodeSendVote(ctx, sstx)
+	if err != nil {
+		return fmt.Errorf("failed to vote: %v", err)
+	}
 
+	return nil
+}
+
+func (ctx *appContext) processWinningTickets(wt WinningTicketsForBlock) {
+	work := false
 	txStrs := make([]string, 0, len(wt.winningTickets))
 	for _, ticket := range wt.winningTickets {
-		txStrs = append(txStrs, ticket.String())
-		for msa := range ctx.userTickets {
-			if !sliceContains(ctx.userTickets[msa].Tickets, ticket) {
-				continue
+		tS := ticket.String()
```

Comment:
This would again would be unnecessary if `ctx.tickets` were a `map[chainhash.Hash]string` instead of a `map[string]string`.